pkg/ctl/ctl_logging.go

```diff
@@ -25,15 +28,16 @@ type ctlLoggingClient struct {
 	logger *zap.SugaredLogger
```

Comment:
You're pulling the logger from the context now so I think this struct field can / should be dropped.

```suggestion
```

Reply:
yes, but without it the tests in `ctl_logging_test.go` won't work because they introspect whether or not the log messages actually made it.

Reply:
If you're pushing the logger into the context I think its better to go all the way and remove all references. Remove the logger from the struct here, remove it from the `WithLogging()` signature and in the test just check that `log.Logger` sees the correct changes when testing.

In general I think explicit dependency injection is better than context stuffing and global vars, but if we're going one way or another I think its better to go all the way if that makes sense.